lib/srv/alpnproxy/protocols.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2020-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpnproxy
+
+// Protocol is the TLS ALPN protocol type.
+type Protocol string
+
+const (
```

Comment:
BTW, there is also `acme-tls/1` APLN protocol used by ACME during TLS ALPN challenge. We need to make sure it's supported as well (it's not present among those defined here currently).